pkg/storage/stores/shipper/compactor/deletion/delete_requests_client.go

```diff
@@ -0,0 +1,174 @@
+package deletion
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log/level"
+
+	"github.com/grafana/loki/pkg/util/log"
+)
+
+const (
+	orgHeaderKey  = "X-Scope-OrgID"
+	getDeletePath = "/loki/api/v1/delete"
+)
+
+type DeleteRequestsClient interface {
+	GetAllDeleteRequestsForUser(ctx context.Context, userID string) ([]DeleteRequest, error)
+	Stop()
+}
+
+type deleteRequestsClient struct {
+	url        string
+	httpClient doer
+	mu         sync.RWMutex
+
+	cache         map[string][]DeleteRequest
+	cacheDuration time.Duration
+
+	stopChan chan struct{}
+}
+
+type doer interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+type DeleteRequestsStoreOption func(c *deleteRequestsClient)
+
+func WithRequestClientCacheDuration(d time.Duration) DeleteRequestsStoreOption {
+	return func(c *deleteRequestsClient) {
+		c.cacheDuration = d
+	}
+}
+
+func NewDeleteRequestsClient(addr string, c doer, opts ...DeleteRequestsStoreOption) (DeleteRequestsClient, error) {
```

Comment:
Instead of taking an HTTP client as a parameter, what do you think about accepting a config?

Reply:
I'd prefer to take the client, if it's not too disruptive. I really like this pattern because it makes if very easy to unit test without worrying about the network: [pkg/storage/stores/shipper/compactor/deletion/delete_requests_client_test.go](https://github.com/grafana/loki/pull/6194/files/51d1535c4d9adf786c76e59f1352dd1d1b6a28dd#diff-1f659c597f41d7bd102f9f32644059a6172892ca7905eb176b0972386f5ed3fe)